service/controller/v3/resource_set.go

```diff
@@ -324,7 +325,12 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return nil, microerror.Mask(err)
 		}
 
-		azureClients, err := client.NewAzureClientSet(config.HostAzureConfig)
+		guestAzureConfig, err := credential.GetAzureConfig(config.K8sClient, obj)
```

Comment:
Isn't the name here a bit misleading? Why is this a guest config? This is not about config of an guest cluster, but about some other Azure subscription, right?

Reply:
Name is somehow misleading but this is aligned with `HostAzureConfig`, those are in fact credential for the Azure API.

I would happily tackle this in a different PR, but I need good naming first. How about `guestAzureCredential` and `hostAzureCredential` ?

Reply:
Ah yeah, naming man. No idea.

Reply:
The returned type is `*client.AzureClientSetConfig`. Pretty clear IMO. If the name was `guestAzureClientSetConfig` I think it would be pretty self explanatory.

Reply:
you said it naming ... ok I'll do a bunch of renaming in a different PR then